lib/pbench/test/unit/server/test_quisby_results.py

```diff
@@ -0,0 +1,116 @@
+from http import HTTPStatus
+from pathlib import Path
+
+from pquisby.lib.post_processing import QuisbyProcessing
+import pytest
+import requests
+
+from pbench.server.api.resources import ApiBase
+from pbench.server.cache_manager import CacheManager
+from pbench.server.database.models.datasets import Dataset, DatasetNotFound
+
+
+class TestQuisbyResults:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, get_token_func):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            pbench_drb_token: Authenticated user token fixture
+        """
+
+        def query_api(
+            dataset: str, user, expected_status: HTTPStatus
+        ) -> requests.Response:
+            try:
+                dataset_id = Dataset.query(name=dataset).resource_id
+            except DatasetNotFound:
+                dataset_id = dataset  # Allow passing deliberately bad value
+            headers = {"authorization": f"bearer {get_token_func(user)}"}
+            response = client.get(
+                f"{server_config.rest_uri}/quisby/{dataset_id}/",
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def mock_find_dataset(self, dataset):
+        class Tarball(object):
+            tarball_path = Path("/dataset/tarball.tar.xz")
+
+            def extract(tarball_path, path):
+                return "CSV_file_as_a_byte_stream"
+
+        return Tarball
+
+    def mock_get_dataset_metadata(self, dataset, key):
+        return {"dataset.metalog.pbench.script": "uperf"}
+
+    def test_get_no_dataset(self, query_get_as):
+        response = query_get_as("nonexistent-dataset", "drb", HTTPStatus.NOT_FOUND)
+        assert response.json == {"message": "Dataset 'nonexistent-dataset' not found"}
+
+    def test_dataset_not_present(self, query_get_as):
+        response = query_get_as("fio_2", "drb", HTTPStatus.NOT_FOUND)
+        assert response.json == {
+            "message": "The dataset tarball named 'random_md5_string4' is not present in the cache manager"
```

Comment:
I think we probably want a better message in the `APIAbort` here: we don't want to expose the "cache manager" concept to the caller. (Or maybe the cache manager shouldn't be unnecessarily advertising itself and should just say dataset not found...)